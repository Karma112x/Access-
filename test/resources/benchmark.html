<!DOCTYPE html>
<button id="button" onclick="logEvent(event)" ondblclick="logEvent(event)">
  Click
</button>
<button
  id="hover"
  onmouseenter="logEvent(event)"
  onmouseleave="logEvent(event)"
>
  Hover
</button>
<input id="input" oncontextmenu="logEvent(event)" onchange="logEvent(event)" />
<input id="input-prefilled" onchange="logEvent(event)" value="test" />
<select id="select" onchange="logEvent(event)">
  <option value=""></option>
  <option value="optionA">Option A</option>
  <option value="optionB">Option B</option>
</select>
<pre id="log"></pre>
<script>
  function logStr(str) {
    log.innerText += str;
  }
  function logEvent(event) {
    logStr(
      '\n' +
        event.type +
        ' targetId=' +
        event.target.id +
        ' button=' +
        event.button +
        ' value=' +
        event.target.value
    );
  }
  logStr(`window dimensions ${window.innerWidth}x${window.innerHeight}`);
  input.addEventListener('contextmenu', (e) => e.preventDefault(), false);
</script>
