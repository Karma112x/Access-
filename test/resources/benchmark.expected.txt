window dimensions 900x700
click targetId=button button=0 value=
click targetId=button button=0 value=
dblclick targetId=button button=0 value=
change targetId=input button=undefined value=test
change targetId=input-prefilled button=undefined value=testSuffix
contextmenu targetId=input button=2 value=test
mouseenter targetId=hover button=0 value=
change targetId=select button=undefined value=optionB
